### ISSUE TITLE
css.async.filter.config load_position => "inview" bug

### ISSUE THREAD

**Issue by ArekZw** [o10n-x/wordpress-css-optimization#38]
When I set "load_position" to "timing", I get error in console:
"Uncaught ReferenceError: K is not defined"

How to fix it?

My JSON:
`{
  "css": {
    "async": {
      "enabled": true,
      "filter": {
        "enabled": true,
        "type": "include",
        "config": [
          {
            "match": "plyr",
            "async": true,
            "load_position": "timing",
            "load_timing": {
              "type": "inview",
              "selector": "#footer",
              "offset": 100
            }
          }
        ]
      }
    }
  }
}`

**Comment by optimalisatie** [comment]
Hi @ArekZw,

Thank you very much for reporting the bug! 

We discovered that the timing modules `responsive.js` and `inview.js` weren't loaded on the basis of Async Filter configuration. The bug has been fixed in `v0.0.93`.

![image](https://user-images.githubusercontent.com/8843669/49545964-76d70a80-f8df-11e8-909b-b3214c63c9b1.png)

### Javascript client script compression innovation

To reduce the size of the client script in the head of the HTML document to the minimum size possible, a special (seemingly) undocumented `--module` feature of Google Closure Compiler is used to compile multiple scripts as independent files that can be stitched together as raw modules without a module loading logic.

The independent modules benefit from Google Closure Compiler's Advanced Optimizations for maximum compression as if it were a single script, while they can be used independently as long as they don't depend on functionalities within an other module for it's execution logic.

In the case of the bug, the timed stylesheet loading or rendering based on the method `element in view` required the extra module `inview.js`. For users who don't use a timing method, all modules related to timing simply aren't loaded to achieve optimal performance for any configuration, while in the same time at maximum complexity, the script as a whole is optimally compressed by Google Closure Compiler, as if the optimization was applied to the whole script.

Example:

##### timed-exec.js (timing controller)

```javascript
function K(a,c,b){switch(a[0]){case 21:B(c);break;case 22:b=1;a[1]&&a[1][15]&&(b=a[1][15]);RAF(c,b);break;case 23:var d,e;a[1]&&a[1][16]&&(d=a[1][16]);a[1]&&a[1][17]&&(e=function(b){setTimeout(b,a[1][17])});E(c,d,e);break;case 24:var f,g;a[1]&&a[1][18]&&(f=a[1][18]);a[1]&&a[1][19]&&(g=a[1][19]);Tb(f,g,c);break;case 20:d="all",a[1]&&a[1][20]&&(d=a[1][20]),Ha(d,c,b)}};
```

##### inview.js (timing module)

```javascript
function Qb(a,c){var b;if((b=(b=a)&&!b.nodeType?b[0]:b)&&1===b.nodeType){b=b.getBoundingClientRect();var d=c,e={};d||(d=0);d=+d||0;e.width=(e.right=b.right+d)-(e.left=b.left-d);e.height=(e.bottom=b.bottom+d)-(e.top=b.top-d);b=e}else b=!1;if(d=!!b&&0<=b.bottom&&0<=b.right)d=ba.clientHeight,e=m.innerHeight,d=b.top<=(d<e?e:d);d&&(d=ba.clientWidth,e=m.innerWidth,d=b.left<=(d<e?e:d));return d}var Rb=[],Sb;
ra("scroll",function(){Rb.length&&(Sb&&clearTimeout(Sb),Sb=setTimeout(function(){for(var a=Rb.splice(0),c=[],b=a.shift(),d=0;b;){if(b&&Qb(b[0],b[1]))b[2]();else c.push(b);b=a.shift();d++}Rb=Rb.concat(c)}))},m);function Tb(a,c,b){B(function(){var d=aa(a);d&&(Qb(d,c)?b():Rb.push([d,c,b]))})}A.inview=Tb;
```

### JSON compression

Besides the Google Closure Compiler script optimization, the JSON configuration is also compressed using an Array index system. Empty values are replaced by `ø` while multiple empty slots in a row are compressed as `ø:11`. Also values are indexed using this method. It could still be improved but as a first version it's saving quite some data.

![image](https://user-images.githubusercontent.com/8843669/49546315-388e1b00-f8e0-11e8-9e91-be7fe4740cbc.png)

**Comment by optimalisatie** [comment]
Thanks a lot for the feedback! It is correct that the bug wasn't fixed correctly for `render_timing`.

The issue has been resolved in `v0.0.94`.

![image](https://user-images.githubusercontent.com/8843669/49570738-c3e6cb00-f937-11e8-9c39-43967066afc9.png)

**Comment by ArekZw** [comment]
Now all works fine, thanks!
